kong/api/routes/upstreams.lua

```diff
@@ -160,4 +103,25 @@ return {
     end
   },
 
+  ["/upstreams/:name_or_id/targets/:target"] = {
+    before = function(self, dao_factory, helpers)
+      crud.find_upstream_by_name_or_id(self, dao_factory, helpers)
+      self.params.upstream_id = self.upstream.id
+    end,
+
+    DELETE = function(self, dao_factory)
+      balancer.clean_history(self.params.upstream_id, dao_factory)
+
+      -- this is just a wrapper around POSTing a new target with weight=0
+      local target  = self.params
```

Comment:
`self.params` represents the request parameters, with modifications from the `before` handler.

I'd consider it cleaner to create a new target table here

```lua
local data, err = dao_factory.targets:insert( {
  target = params.target,
  upstream_id = self.upstream.id,
  weight = 0,
})
```
Then line 109 above can be dropped as well.